lib-rt/CPy.h

```diff
@@ -817,6 +817,23 @@ static int CPyDict_UpdateFromSeq(PyObject *dict, PyObject *stuff) {
     }
 }
 
+PyObject *CPyDict_FromSeq(PyObject *obj) {
+    if (PyDict_Check(obj)) {
```

Comment:
What if this is a non-dictionary mapping? Should we then call `items()` perhaps? (I didn't look at how `dict` decides whether the argument is a mapping or not.)